[Docs] Refactoring the `PATH` Developer Guide docs (#137)
buildwithgrove/path: docusaurus/docs/develop/path/walkthrough.md

- [Running PATH](#running-path)
  - [1. Tilt Mode (Recommended)](#1-tilt-mode-recommended)
  - [2. Standalone Binary Mode](#2-standalone-binary-mode)
- [2. Protocol Configuration](#2-protocol-configuration)
  - [2.1 Shannon Protocol Configs](#21-shannon-protocol-configs)
  - [2.2 Morse Protocol Configs](#22-morse-protocol-configs)
- [3. Envoy Proxy Configuration](#3-envoy-proxy-configuration)
  - [3.1 Configuring Relay Authorization](#31-configuring-relay-authorization)
- [4. Run the `PATH` Gateway](#4-run-the-path-gateway)
  - [4.1 View `PATH` Resources in Tilt](#41-view-path-resources-in-tilt)
  - [4.2 Wait for the `PATH` stack to initialize](#42-wait-for-the-path-stack-to-initialize)
- [5. Send a Relay](#5-send-a-relay)
  - [5.1 Envoy Proxy Mode with Static Key Authorization](#51-envoy-proxy-mode-with-static-key-authorization)
  - [5.2 Envoy Proxy without any Authorization](#52-envoy-proxy-without-any-authorization)
  - [5.3 Standalone mode without any Authorization](#53-standalone-mode-without-any-authorization)
- [E2E Tests](#e2e-tests)

## Running PATH

PATH offers two deployment modes for local development:

> ### 1. Tilt Mode (Recommended)

Reviewer: Any reason we can't keep the Tilt logo linking to the docs in here? I thought it added a nice bit of aesthetic variation and interest to the docs.

```suggestion
### 1. Tilt Mode (Recommended)

<div align="center">
  <a href="https://docs.tilt.dev/">
    <img src="https://blog.tilt.dev/assets/img/blog-default-preview.png" alt="Tilt logo" width="200"/>
  <p><b>Tilt Documentation</b></p>
  </a>
</div>
```

Runs PATH with full functionality in a local Kubernetes cluster:

- Routes through Envoy Proxy (port `3070`)
- Enables authorization, rate limiting, and service aliasing
- Uses configs from `/local` directory with Kind + Tilt

Review these configurations:

- [**Tiltfile**](https://github.com/buildwithgrove/path/tree/main/Tiltfile): Tiltfile config file in the PATH repository
- [**Values file**](https://github.com/buildwithgrove/path/tree/main/local/path/config/path-values.yaml): Values file for the local cluster in the PATH repository

> - [**Tilt Documentation**](https://docs.tilt.dev/): Tilt documentation

Reviewer: Related to above comment re: linking to Tilt docs using their logo.
```suggestion
```

for details on configuring a Morse gateway.

If you are comfortable updating the config file manually, then:

```sh
# Create ./e2e/.morse.config.yaml
make prepare_morse_e2e_config
# Update it manually

# Copy it to ./local/path/config/.config.yaml
make copy_morse_config_to_local

# Copy it to ./bin/config/.config.yaml
make copy_morse_config_to_bin
```

## 3. Envoy Proxy Configuration

See the [Envoy Walkthrough](./../envoy/walkthrough.md) for all the details
on running and configuring Envoy.

To get up and running quickly, run:

```sh
make init_envoy
```

**We recommend choosing Option 2 (no authorization) for now as a simpler starting point.**

If you wish to use an `0Auth` provider _([for example Auth0](https://auth0.com))_ to enable
authorizing requests using an issued JWT, you will need to provide the `AUTH_DOMAIN` and
`AUTH_AUDIENCE` values to substitute the sensitive variables in the `envoy.yaml` file.

If you do not wish to use an OAuth provider, simply answer `no` when prompted.
This will allow authorizing requests with a static API key only.

### 3.1 Configuring Relay Authorization

:::tip

> Saving `.gateway-endpoints.yaml` triggers a hot-reload of PADS (PATH Auth Data Server) in Tilt.

Reviewer: ```suggestion
Saving `.gateway-endpoints.yaml` will automatically stream the updated file contents to PADS (PATH Auth Data Server) in Tilt; there is no need to restart PADS.
```
